src/components/common/Extensions/MainNavigation/partials/Nav/Nav.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { FC, useCallback, useState } from 'react';
+import clsx from 'clsx';
+import { useIntl } from 'react-intl';
+import { AnimatePresence, motion } from 'framer-motion';
+import { NavItemProps, NavProps } from './types';
+import Icon from '~shared/Icon';
+import styles from './Nav.module.css';
+import SubMenu from '../SubMenu';
+import { useMobile } from '~hooks';
+
+const displayName = 'common.Extensions.MainNavigation.partials.Nav';
+
+const NavItem: FC<NavItemProps> = ({ item }) => {
+  const { formatMessage } = useIntl();
+  const isMobile = useMobile();
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const toggleItem = useCallback(() => {
+    setIsOpen(!isOpen);
+
+    if (item.onToggle) {
+      item.onToggle(!isOpen);
+    }
+  }, [isOpen, item]);
+
+  const navLink = (
+    <a className={styles.navLink} href={item.href}>
+      {formatMessage({
+        id: `mainNavItem.${item.label}`,
+        defaultMessage: `${item.label}`,
+      })}
+    </a>
+  );
+
+  return isMobile ? (
+    <>
+      {item.href ? (
+        navLink
+      ) : (
+        <button type="button" className={clsx(styles.navLink, { 'text-blue-400': isOpen })} onClick={toggleItem}>
+          {formatMessage({
+            id: `mainNavItem.${item.label}`,
+            defaultMessage: `${item.label}`,
+          })}
+          {item.subMenu && (
+            <span className={clsx('flex ml-2.5 transition-transform duration-normal', { 'rotate-180': isOpen })}>
+              <Icon name="caret-down" appearance={{ size: 'extraTiny' }} />
+            </span>
+          )}
+        </button>
+      )}
+      {item.subMenu && (
+        <div className={styles.subMenuMobile}>
+          <AnimatePresence>
+            {isOpen && (
+              <motion.div
+                key="accordion-content"
+                initial={{ opacity: 0, height: 0 }}
+                animate={{ opacity: 1, height: 'auto' }}
+                exit={{ opacity: 0, height: 0 }}
+                transition={{ duration: 0.4, ease: 'easeOut' }}
```

Comment:
I think it's better to use variants here ![image](https://github.com/JoinColony/colonyCDapp/assets/76940796/d19166cc-5878-421e-ab6d-5753604a48ea)